consensus/polybft/state_sync_manager.go

```diff
@@ -381,6 +385,32 @@ func (s *stateSyncManager) PostBlock(req *PostBlockRequest) error {
 	return nil
 }
 
+// GetStateSyncProof returns the proof for the state sync
+func (s *stateSyncManager) GetStateSyncProof(stateSyncID uint64) (*types.StateSyncProof, error) {
+	proof, err := s.state.getStateSyncProof(stateSyncID)
+	if err != nil {
+		return nil, fmt.Errorf("cannot get state sync proof for StateSync id %d: %w", stateSyncID, err)
+	}
+
+	if proof == nil {
+		// check if we might've missed a commitment. if it is so, we didn't build proofs for it while syncing
+		// if we are all synced up, commitment will be saved through PostBlock, but we wont have proofs,
+		// so we will build them now and save them to db so that we have proofs for missed commitment
+		commitment, err := s.state.getCommitmentForStateSync(stateSyncID)
+		if err != nil {
+			return nil, fmt.Errorf("cannot find commitment for StateSync id %d: %w", stateSyncID, err)
+		}
+
+		if err := s.buildProofs(commitment.Message); err != nil {
+			return nil, fmt.Errorf("cannot build proofs for commitment for StateSync id %d: %w", stateSyncID, err)
+		}
+
+		proof, err = s.state.getStateSyncProof(stateSyncID)
```

Comment:
Maybe check the error here to return a descriptive one, and return the proof if there's no error?

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1141/commits/b58c1f2b790131f7c395ce98c05d19935f7b20b8